File: src/db-toolkit/connectors/factory.py
"""Connector factory for creating database connectors."""

from typing import Dict, Type
from .base import BaseConnector
from .postgresql import PostgreSQLConnector
from .mysql import MySQLConnector
from .sqlite import SQLiteConnector
from .mongodb import MongoDBConnector
from ..core.models import DatabaseType


class ConnectorFactory:
    """Factory for creating database connectors."""
    
    _connectors: Dict[DatabaseType, Type[BaseConnector]] = {
        DatabaseType.POSTGRESQL: PostgreSQLConnector,
        DatabaseType.MYSQL: MySQLConnector,
        DatabaseType.SQLITE: SQLiteConnector,
        DatabaseType.MONGODB: MongoDBConnector,
    }
    
    @classmethod
    def create_connector(cls, db_type: DatabaseType) -> BaseConnector:
        """Create connector instance for database type."""
        connector_class = cls._connectors.get(db_type)
        if not connector_class:
            raise ValueError(f"Unsupported database type: {db_type}")
        return connector_class()
    
    @classmethod
    def get_supported_types(cls) -> list[DatabaseType]:
        """Get list of supported database types."""
        return list(cls._connectors.keys())

File: src/db-toolkit/electron/backend/connectors/factory.js
/**
 * Connector factory for creating database connectors.
 */

const PostgreSQLConnector = require('./postgresql');
const MySQLConnector = require('./mysql');
const MariaDBConnector = require('./mariadb');
const SQLiteConnector = require('./sqlite');
const MongoDBConnector = require('./mongodb');

const connectors = {
  postgresql: PostgreSQLConnector,
  mysql: MySQLConnector,
  mariadb: MariaDBConnector,
  sqlite: SQLiteConnector,
  mongodb: MongoDBConnector,
};

class ConnectorFactory {
  static createConnector(dbType) {
    const ConnectorClass = connectors[dbType.toLowerCase()];
    if (!ConnectorClass) {
      throw new Error(`Unsupported database type: ${dbType}`);
    }
    return new ConnectorClass();
  }

  static getSupportedTypes() {
    return Object.keys(connectors);
  }
}

module.exports = ConnectorFactory;
